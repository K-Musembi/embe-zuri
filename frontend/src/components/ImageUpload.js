import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
import { API_URL } from "../config";

const ImageUpload = ({ onResult }) => {
    const [file, setFile] = useState(null);
    const navigate = useNavigate();

    const handleFileChange = (event) => {
        setFile(event.target.files[0]);  // when event occurs, set file
    };

    const handleUpload = async () => {
        if (!file) return;  // return if no file is selected

        const data = new FormData();
        data.append("image", file);

        try {
            const response = await fetch(`${API_URL}/upload`, {
                method: "POST",
                body: data,
            });
            if (response.ok) {
                const result = await response.json();
                onResult(result);  // pass result to parent component
                navigate("/result");
            }
        } catch (err) {
            console.error("Upload failed:", err);
        }
    };  
    return (
        <div className="image-upload-container">
            <h2>Upload an image</h2>
            <label htmlFor="file-upload" className="custom-file-upload">
                Choose File
            </label>
            <input id="file-upload" type="file" accept="image/*" style={{ display: 'none' }} onChange={handleFileChange} />
            <button onClick={handleUpload}>Upload</button>
        </div>
    );
};

export default ImageUpload;
