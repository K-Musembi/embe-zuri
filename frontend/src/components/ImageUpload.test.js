import React from 'react';
import renderer from 'react-test-renderer';
import ImageUpload from './ImageUpload';
import { MemoryRouter } from 'react-router-dom';

describe('ImageUpload component snapshot test', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
        <MemoryRouter>
          <ImageUpload />
        </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

global.fetch = jest.fn(() =>
    Promise.resolve({
      ok: true,
      json: () => Promise.resolve({ message: 'Success' })
    })
  );
