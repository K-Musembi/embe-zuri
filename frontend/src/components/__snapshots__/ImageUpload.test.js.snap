// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageUpload component snapshot test renders correctly 1`] = `
<div
  className="image-upload-container"
>
  <h2>
    Upload a mango leaf image
  </h2>
  <input
    accept="image/*"
    id="file-upload"
    onChange={[Function]}
    type="file"
  />
  <button
    disabled={true}
    onClick={[Function]}
  >
    Upload
  </button>
</div>
`;
